strawberry/field.py

```diff
@@ -9,292 +8,305 @@
     Callable,
     Dict,
     List,
-    Mapping,
     Optional,
     Sequence,
     Type,
-    TypeVar,
     Union,
+    cast,
     overload,
 )
 
 from backports.cached_property import cached_property
-from typing_extensions import Literal
+from typing_extensions import Literal, T
 
 from strawberry.annotation import StrawberryAnnotation
 from strawberry.arguments import StrawberryArgument
-from strawberry.exceptions import InvalidDefaultFactoryError, InvalidFieldArgument
-from strawberry.type import StrawberryType, StrawberryTypeVar
+from strawberry.exceptions import (
+    FieldWithResolverAndDefaultFactoryError,
+    FieldWithResolverAndDefaultValueError,
+    InvalidDefaultFactoryError,
+    InvalidFieldArgument,
+    MissingFieldAnnotationError,
+    MissingReturnAnnotationError,
+    PrivateStrawberryFieldError,
+)
+from strawberry.type import StrawberryType
 from strawberry.types.info import Info
 from strawberry.union import StrawberryUnion
 from strawberry.unset import UNSET
 
 from .permission import BasePermission
-from .types.fields.resolver import StrawberryResolver
+from .private import is_private
+from .types.fields.resolver import StrawberryResolver, resolveable
 
 
 if TYPE_CHECKING:
-    from .object_type import TypeDefinition
-
+    pass
 
 _RESOLVER_TYPE = Union[StrawberryResolver, Callable, staticmethod, classmethod]
 
-
 UNRESOLVED = object()
 
 
-class StrawberryField(dataclasses.Field):
-    python_name: str
-    default_resolver: Callable[[Any, str], object] = getattr
-
-    def __init__(
+@dataclasses.dataclass()
+class StrawberryField(StrawberryType):
```

Comment:
I'm not sure this should extend StrawberryType /cc @BryceBeagle

Reply:
This is also for `validate()` see https://github.com/strawberry-graphql/strawberry/pull/2117#issuecomment-1231474766